### ISSUE TITLE
Threading (test and possible change)

### ISSUE THREAD

**Issue by janesth** [ngwn-studio/f1-udp#15]
**Current implementation in branch ``master``**
*Receiver.java*
```
try {
            DatagramSocket socket = new DatagramSocket(port);
            byte[] buffer = new byte[MAX_BUFFER];
            DatagramPacket packet = new DatagramPacket(buffer, buffer.length);

            while(is_running) {
                socket.receive(packet);
                new Thread(new Runnable() {
                    @Override
                    public void run() {
                        controller.handlePacket(buffer);
                    }
                }).run();

                packet.setLength(buffer.length);
            }

        } catch(IOException ex) {
            System.out.println("Check port: '" + port + "' or packet.");
        }
```

**Implementation in branch ``14-view-selector``**
*DataController.java*
```
public DataController() {
        receiver = new Receiver(this);
        new Thread(receiver).start();
        }
    }
```

**Issue:**
Testing the port on mac with ``nc -vnzu 127.0.0.1 4445`` in debug mode activates a breakpoint in ``Datacontroller.handlePacket`` only once. This indicates that the thread is ended after only one run. The implementation in master can not be used because it causes a freeze in the application.

- Check in-game if thread continues to run
- If it does not continue:
  - Look into other ways to keep the thread going
  - Other options besides threads...?

**Comment by janesth** [comment]
In case branch ``14-view-selector`` hasn't been merged, please use it as source branch. If it has merged, use branch ``develop`` as usual.

**Comment by janesth** [comment]
changed logic in commit b80d0b7473af9eb23f989859c3d936b3ac68269b - needs to be tested in-game.

**Comment by janesth** [comment]
I checked this again and from my understanding this works now. I'm leaving the stack traces call in DataController.java:17 in, because it's interesting for testing purposes. I'll remove it in another ticket, probably #18
